Latest master branch doesn't build against 4.05.0

Tested under Ubuntu 16.04 64bit with OCaml 4.05.0 installed with opam after pinning with `--dev-repo`:
```
#=== ERROR while installing ocamlformat.v0.2 ==================================#
# opam-version 1.2.2        
# os           linux        
# command      make
# path         /home/hcarty/.opam/4.05.0/build/ocamlformat.v0.2
# compiler     4.05.0
# exit-code    2
# env-file     /home/hcarty/.opam/4.05.0/build/ocamlformat.v0.2/ocamlformat-76215-a71bb4.env
# stdout-file  /home/hcarty/.opam/4.05.0/build/ocamlformat.v0.2/ocamlformat-76215-a71bb4.out
# stderr-file  /home/hcarty/.opam/4.05.0/build/ocamlformat.v0.2/ocamlformat-76215-a71bb4.err
### stdout ###
# sed -e "s|@OPAM_SWITCH[@]|$(opam switch show)|g" src/jbuild-workspace.in > src/jbuild-workspace
# patching file src/Version.ml
# jbuilder build --root=src -j 4 ocamlformat.exe
# Makefile:35: recipe for target 'exe' failed
### stderr ###
# Entering directory '/home/hcarty/.opam/4.05.0/build/ocamlformat.v0.2/src'
# [...]
#     ocamlopt Ast.{cmx,o} [opt]
#     ocamlopt Translation_unit.{cmx,o} [dbg] (exit 2)
# (cd _build/dbg && /home/hcarty/.opam/4.05.0/bin/ocamlopt.opt -w +a-4-6-9-40-41-42-44-45-48@50 -strict-formats -strict-sequence -principal -short-paths -bin-annot -keep-docs -unboxed-types -open Import -g -opaque -w -a -I /home/hcarty/.opam/4.05.0/lib/base -I /home/hcarty/.opam/4.05.0/lib/base/caml -I /home/hcarty/.opam/4.05.0/lib/base/shadow_stdlib -I /home/hcarty/.opam/4.05.0/lib/bytes -I /home/hcarty/.opam/4.05.0/lib/cmdliner -I /home/hcarty/.opam/4.05.0/lib/ocaml -I /home/hcarty/.opam/4.05.0/lib/ocaml-migrate-parsetree -I /home/hcarty/.opam/4.05.0/lib/ocaml/compiler-libs -I /home/hcarty/.opam/4.05.0/lib/ocamlformat_support -I /home/hcarty/.opam/4.05.0/lib/result -I /home/hcarty/.opam/4.05.0/lib/sexplib/0 -I /home/hcarty/.opam/4.05.0/lib/stdio -I import -no-alias-deps -I . -o Translation_unit.cmx -c -impl Translation_unit.ml)
# File "Translation_unit.ml", line 133, characters 4-19:
# Error: The constructor Warnings.Errors expects 1 argument(s),
#        but is applied here to 0 argument(s)
# 
# Waiting for 1 job to finish.
# make: *** [exe] Error 1
```

--- a/src/Translation_unit.ml
+++ b/src/Translation_unit.ml
@@ -129,8 +129,12 @@ let parse_print (XUnit xunit) (conf: Conf.t) iname ifile ic ofile =
       | Inplace _, _ -> Unix.unlink tmp
   in
   let source = In_channel.with_file ifile ~f:In_channel.input_all in
-  try parse_print_ 1 source ifile ic with
-  | Warnings.Errors -> Caml.exit 1
+  (* NOTE: Warning 28 is suppressed due to a difference in exception
+     constructor arity between OCaml versions.  See this
+     ocaml-migrate-parsetree issue for potential future mitigation.
+     https://github.com/ocaml-ppx/ocaml-migrate-parsetree/issues/34 *)
+  try[@ocaml.warning "-28"] parse_print_ 1 source ifile ic with
+  | Warnings.Errors _ -> Caml.exit 1
   | Syntaxerr.Error _ as exc ->
       Location.report_exception Caml.Format.err_formatter exc ;
       Caml.exit 1

